ec2/networkinterface.py

```diff
@@ -0,0 +1,213 @@
+       ctx.instance.runtime_properties.get('aws_resource_id'):
+        ec2_client = connection.EC2ConnectionClient().client()
+        try:
+            # Delete the Network Interface
+            ec2_client.delete_network_interface(
+                ctx.instance.runtime_properties.get('aws_resource_id'))
+        except (boto.exception.EC2ResponseError,
+                boto.exception.BotoServerError) as ex:
+            raise NonRecoverableError('{0}'.format(str(ex)))
+
+
+@operation
+def attach_elastic_ip(**_):
+    '''Attaches an Elastic IP to this Elastic Network Interface'''
+    ec2_client = connection.EC2ConnectionClient().client()
+    # Get the Elastic Netowrk Interface runtime properties
+    eni_props = ctx.source.instance.runtime_properties
+    # Attempt to associate the EIP with the ENI
+    try:
+        _get_address_object(
+            ec2_client,
+            ctx.target.instance.runtime_properties.get('aws_resource_id')
+        ).associate(
+            network_interface_id=eni_props.get('aws_resource_id'),
+            private_ip_address=eni_props.get('aws_private_ip_address'))
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError) as ex:
+        raise NonRecoverableError('{0}'.format(str(ex)))
+
+
+@operation
+def detach_elastic_ip(**_):
+    '''Detaches an Elastic IP from this Elastic Network Interface'''
+    ec2_client = connection.EC2ConnectionClient().client()
+    # Attempt to disassociate the EIP from the ENI
+    try:
+        _get_address_object(
+            ec2_client,
+            ctx.target.instance.runtime_properties.get('aws_resource_id')
+        ).disassociate()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError) as ex:
+        raise NonRecoverableError('{0}'.format(str(ex)))
+
+
+def _find_attached_subnet():
+    '''Finds a Subnet attached via relationship'''
+    for rel in ctx.instance.relationships:
+        if SUBNET_RELATIONSHIP in rel.type_hierarchy and \
+           SUBNET_TYPE in rel.target.node.type_hierarchy:
+            return rel.target
+    return None
+
+
+def _find_attached_security_groups():
+    '''Finds all attached Security Groups via relationships'''
+    return [
+        x.instance.runtime_properties['aws_resource_id']
+        for x in [
+            rel.target for rel in ctx.instance.relationships
+            if SECURITY_GROUP_RELATIONSHIP in rel.type_hierarchy and
+            SECURITY_GROUP_TYPE in rel.target.node.type_hierarchy
+        ]
+        if 'aws_resource_id' in x.instance.runtime_properties
+    ]
+
+
+def _get_address_object(ec2_client, eip_ip):
+    '''Gets an Address (Elastic IP) object from AWS'''
+    eip_objs = ec2_client.get_all_addresses(addresses=[eip_ip])
+    if not isinstance(eip_objs, list) or len(eip_objs) < 1:
+        raise NonRecoverableError(
+            'Could not get Address object')
+    return eip_objs[0]
+
+
+def _get_network_interface_object(ec2_client, eni_id):
```

Comment:
use the [established pattern](https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/ec2/elasticip.py#L322) of getting boto obj by id